academy/exchange/__init__.py

```diff
@@ -54,6 +57,27 @@ class MailboxStatus(enum.Enum):
     """Mailbox was terminated and no longer accepts messages."""
 
 
+class RegistrationInfo(Protocol[BehaviorT]):
```

Comment:
Will we ever need to have a similar class for user registrations? If not `AgentRegistration` may be a better name.